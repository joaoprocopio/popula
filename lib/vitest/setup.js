import { config } from '@vue/test-utils'

import { vuetify } from '~/plugins/vuetify'

config.global.plugins = [vuetify]
