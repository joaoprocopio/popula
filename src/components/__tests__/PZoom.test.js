import { mount } from '@vue/test-utils'

import { describe, expect, test } from 'vitest'

import { PZoom } from '..'

const render = (config = {}) => mount(PZoom, config)

describe('PZoom', () => {
  const wrapper = render()

  test('Should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('On click should emit "zoomOut"', async () => {
    await wrapper.get('[data-test-id="zoomOut"]').trigger('click')

    expect(wrapper.emitted().zoomOut).toBeTruthy()
  })

  test('On click should emit "zoomIn"', async () => {
    await wrapper.get('[data-test-id="zoomIn"]').trigger('click')

    expect(wrapper.emitted().zoomIn).toBeTruthy()
  })
})
