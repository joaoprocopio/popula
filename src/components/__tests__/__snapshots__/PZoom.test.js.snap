// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PZoom > Should render correctly 1`] = `
"<div class=\\"v-btn-group v-theme--light v-btn-group--density-default elevation-0\\"><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--icon v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-elevated\\" style=\\"height: auto;\\" data-test-id=\\"zoomOut\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"material-icons v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\">remove</i></span>
    <!---->
    <!---->
  </button><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--icon v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-elevated\\" style=\\"height: auto;\\" data-test-id=\\"zoomIn\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"material-icons v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\">add</i></span>
    <!---->
    <!---->
  </button></div>"
`;
